pagopa/io-app
# feat(IT Wallet): [SIW-1470] Add IT Wallet skeumorphic credential cards

## Short description
This PR implements the skeumorphic credential card component and its MDL variant.

> [!NOTE]  
> This PR does not include yet the final skeumorphic MDL credential assets, which will be added later.

## List of changes proposed in this pull request
- Added MDL skeumorphic card assets
- Added MDL mock data
- Added `ItwSkeumorphicCard` component, which includes:
  - `FlippableCard` component, which allows to render a double sided component and flip it with an animation showing both sides
  - `CardBackground` component, which renders the correct card background based on credential type and card side
  - `ClaimLabel` component, which is a custom text component to correctly render the credential data
  - `CardClaim` component, which renders the correct component based on the provided claim type using a runtime decoder. For example: if the claim is a date it renders a `ClaimLabel` component with the formatted date, or if the claim is an image it renders an `AnimatedImage` component with the claim content
  - `CardClaimRenderer` component, which allows to render a claim using a custom decoder and renderer function
  - `CardData` component, which renders the correct credential data in the correct position and size, based on the device's screen size
- Added tests
- Added `ItwSkeumorphicCard` playground within the IT Wallet's playgrounds
- Added new type helpers: `Only<T, U>` and `Either<T, U>`

## How to test
Using devices with different screen sizes, go to **Profile > Playground > IT Wallet** and check that the skeumorphic credential is correctly rendered.



## Preview

|iPhone 13 Mini | iPhone 15 | iPhone 15 Pro Max | 
| -- | --- | --- |
| <video src="https://github.com/user-attachments/assets/449c08ef-6adf-4f7b-b4d1-1ac6e27fb2c3" /> | <video src="https://github.com/user-attachments/assets/b09d4ae9-bc6d-4042-8f13-c304b0588d1e" /> | <video src="https://github.com/user-attachments/assets/cabad8aa-a78e-4fe2-acf9-3808243e88f0" /> |
